go/vm/evmzero/ct.go

```diff
@@ -39,12 +40,9 @@ func NewConformanceTestingTarget() ct.Evm {
 type ctAdapter struct{}
 
 func (a ctAdapter) StepN(state *st.State, numSteps int) (*st.State, error) {
-	// Hack: Special handling for unknown revision, because evmzero cannot represent an invalid revision.
-	// So we mark the status as failed already.
-	// TODO: Fix this once we add full revision support to the CT and evmzero.
-	if state.Revision > common.R10_London {
+	if state.Revision > common.Revision(evmc.NewestSupportedRevision) {
```

Comment:
This line implicitly assumes that the revisions defined in the `vm` package encoded the same way as the revisions in the CT's `common` package.

You could first run the `utils.ToVmParameters(state)` function below and then ask for the revision in the parameters. This revision is of the correct type, and the util function is handling the conversion for you.

The other CT adaptors have the same issue.